### PR TITLE
feat: adds new `<ShowcaseProfile>` component to showcase profiles like GitHub users, repositories or any other profile

### PR DESCRIPTION
**Pull request by julien-deramond**
**Describe the pull request**

This PR supersedes https://github.com/HiDeoo/starlight-showcases/pull/5.

The original idea was to provide a GitHub component for users. But we agreed that having a Profile component was more generic and useful.

The idea is to have a global component named `<ShowcaseProfile>`, that could be used for any profiles, but also with sufficient flexibility to render a GitHub profile/repository.

**Why**

As discussed together, this could be an interesting new component in `starlight-showcases`.

**How**

`<ShowcaseProfile>` works like `<ShowcaseText>` and accept an array of entries. Internally, it uses the `<ShowcaseProfileCard>` component to render each entry.
Its rendering is a vertical box of cards with a gap between them, with the same approach as in `<ShowcaseTwitter>`.

Each entry has props:
- `description` (optional): the description of the profile
- `href`:  the link of the profile
- `name`: the name of the profile
- `picture`: the picture of the profile

Technically, `<ShowcaseProfileCard>` is an adapted version of Starlight `<LinkCard>` component. It basically replaces the icon by a picture defined as a parameter (this image being rounded in CSS).

In the original version, I defined a different rendering for small breakpoints, but it was a bit overkill to do that here, so I abandoned the idea.

**Screenshots**
Screenshot 2024-07-17 at 15 21 23
Screenshot 2024-07-17 at 15 21 16

---

**Demonstrate how to make GitHub profiles**

This component is really generic. However, one enhancement (maybe later) could be to show in the documentation how to get automatically the avatar and the description for GitHub profiles, a quick "how-to" to show how it can be done easily.

**How to retrieve description and avatar automatically?**

And then call the GitHub public API when the rendering is built:
- GET https://api.github.com/users/ghost (user)
- GET https://api.github.com/repos/hideoo/starlight-showcases (project)

Please note that the primary rate limit for unauthenticated requests is 60 requests per hour. (source: https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28)

### CODE REVIEW

File: docs/src/content/docs/demo.mdx [hunk 1]
```diff
@@ -151,3 +151,34 @@ import { ShowcaseQuote } from 'starlight-showcases'
...
+      href: 'https://github.com/HiDeoo/starlight-showcases',
+      name: '<span style="font-weight: normal">HiDeoo/</span><br>starlight-showcases',
+      description: 'Discover the Starlight Showcases components in action.',
+      picture: 'https://avatars.githubusercontent.com/u/494699?v=4',
+    },
+    {
+      href: 'https://github.com/ghost',
+      name: 'ghost',
+      description:
+        "Hi, I'm @ghost! I take the place of user accounts that have been deleted. :ghost:",
```
Comment (julien-deramond) [inline docs/src/content/docs/demo.mdx:180]: 3 possibilities here with different levels of complexity:
- Drop the ":ghost:" text from the description, as it's not rendered
- Replace the ":ghost:" text with the emoji
- Be able to interpret ":{image}:" in descriptions

File: packages/starlight-showcases/components/ShowcaseProfileCard.astro [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+const { name, description, picture, ...attributes } = Astro.props
+---
+
+<div class="starlight-showcases-profile-card">
+  <span class="sl-flex stack">
+    <a {...attributes}>
+      <span class="title" set:html={name} />
+    </a>
+    {description && <span class="description" set:html={description} />}
```
Comment (julien-deramond) [inline packages/starlight-showcases/components/ShowcaseProfileCard.astro:21]: For "stack" and "description", I didn't if you preferred to keep them with their original `<LinkCard>` naming or use a `starlight-showcases-profile-*` naming.

File: docs/src/content/docs/demo.mdx (hunk 1 above)
Comment (HiDeoo) [reply docs/src/content/docs/demo.mdx]: Or, considering users would be the one providing data and optionally fetching it, we can also let them massage the data which could be coming from GitHub and uses this syntax or somewhere else which uses a totally different syntax.

File: docs/src/content/docs/components/profiles.mdx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+The name of the profile to display.
+The value can include HTML for formatting and additional styling.
+
+### `href`
+
+**required**  
+**type:** `string`
+
+The links to the profile page the profile represents.
```
Comment (julien-deramond) [inline docs/src/content/docs/components/profiles.mdx:96]:
```suggestion
The link to the profile page the profile represents.
```

**Review by julien-deramond — COMMENTED** [review]
Thanks for the modifications. They're great! 🙌
I've looked at them carefully, and everything looks fine!
I've slightly updated the description to remove the remaining sub-tasks.

It's not useful in this PR, but as mentioned in the description, it might be useful at some point, somewhere (maybe just a blog post or something) to show how one can retrieve GitHub avatar and description automatically with the token-less GitHub API.
If you think it's something interesting, I could prepare another PR in the future, as I'll probably do it in my own website later.

**Review by HiDeoo — APPROVED** [review]
Thanks a lot for the amazing work and pull request. Very thorough and well executed as usual 👍 

This is looking really great and I think it's time to ship this new component :rocket:

**Pull request merged**